File: LeetCode_Algorithm/590(N-ary Tree Postorder Traversal)/590.py
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children

class Solution:
    def postorder(self, root: 'Node') -> list[int]:
        # sol1 53ms 58% 16.2MB 41%
        self.ans = []
        self.solve(root)
        return self.ans
    def solve(self, node):
        if not node:
            return 
        for child in node.children:
            self.solve(child)
        self.ans.append(node.val)

File: LeetCode_Algorithm/590(N-ary Tree Postorder Traversal)/590.js
var solve = (node) => {
  if (!node)
    return ;
  for (let child of node.children)
    solve(child)
  ans.push(node.val)
}
var postorder = function(root) {
  // sol1 79ms 76% 46MB 19%
  ans = []
  solve(root)
  return ans  
};